internal/host/common.go

```diff
@@ -180,6 +180,27 @@ func GetHostnameAndEffectiveRoleByIP(ip string, hosts []*models.Host) (string, m
 	return "", "", fmt.Errorf("host with IP %s not found in inventory", ip)
 }
 
+func FindHostByID(hostID strfmt.UUID, hosts []*models.Host) *models.Host {
+	for _, h := range hosts {
```

Comment:
Not crucial, but just for clarity can be simplified to something like:
```
return funk.Find(hosts, func(host *models.Host) bool {
    return host.ID == hostID
})
```

Reply:
There is cast on the return value only if it is not nil.  So doesn't seem simpler.